css/products.css

```diff
@@ -0,0 +1,199 @@
+/* single-column layout */
+body {
+    display: grid;
+    grid-template-areas: "header"
+        "nav"
+        "main"
+        "footer";
+    grid-template-rows: auto var(--size-15) 1fr auto;
+    grid-template-columns: 1fr;
+    min-height: 100vh;
+}
+
+/* header */
+.header {
+    grid-area: header;
+}
+
+/* aside-navbar for filters*/
+.aside {
+    grid-area: nav;
+    background-color: var(--dark-grey-color);
+    position: relative;
+}
+
+.aside-navbar {
+    display: flex;
+    flex-direction: column;
+    background-color: var(--white);
+    transform: translateX(-100%);
+    transition: transform 560ms;
+    padding-bottom: var(--size-13);
+}
+
+.aside-navbar.show {
+    transform: translateX(0);
+}
+
+.beautify-aside {
+    z-index: 5;
+    background-color: var(--aside-beautify-color);
+    min-height: 100vh;
+}
+
+.aside-hamburger {
+    background-color: transparent;
+    border: 0;
+    font-size: var(--size-10);
+    cursor: pointer;
+    /* transition: transform 560ms; */
+    position: absolute;
+    top: 0;
+    right: 0;
+}
+
+.aside-hamburger .material-icons {
+    font-size: var(--size-13);
+}
+
+.aside-hamburger .material-icons:hover {
+    transform: scale(1.5);
+}
+
+/* aside-navabr card-filters */
+.card-with-no-border{
+    border: 0;
+}
+
+.card-heading-dflex{
+display: flex;
+justify-content: space-between;
+align-items: center;
+}
+
+.card-title-m-0,
+.card-subtitle-m-0{
+    margin: 0;
+}
+
+.card-footer-dir-col{
+    flex-direction: column;
+    align-content: flex-start;
+    padding:0 var(--size-9) var(--size-9) var(--size-9);
+    border-top: 0;
+    row-gap: var(--size-6);
+}
+
+.card-filter-category{
+    display: flex;
+    align-items: center;
+    column-gap: var(--size-6);
+    min-width: 60%;
+    max-width: max-content;
+}
+
+.card-filter-category > input{
+    width: var(--size-11);
+    height: var(--size-11);
+}
+
+.clear-all-filters{
+    cursor: pointer;
+}
+
+.clear-all-filters:hover{
+    color: var(--logo-color);
+}
+
+/* main-content*/
+.main {
+    grid-area: main;
+    background-color: var(--grey-color);
+}
+
+/* main-content cards */
+.cards-container {
+    margin: 2em;
+    display: grid;
+    row-gap: var(--size-11);
+    grid-template-columns: repeat(auto-fill,minmax(380px, 1fr));
+  }
+
+.cards-container .card{
+    width: 380px;
```

Comment:
can use rems here

Reply:
I will the implement the suggested change, I have used fixed units as part of my expirement.